libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -68,7 +68,9 @@ void SyntaxChecker::endVisit(SourceUnit const& _sourceUnit)
 				string(";\"");
 
 		// when reporting the warning, print the source name only
-		m_errorReporter.warning(3420_error, {-1, -1, _sourceUnit.location().source}, errorString);
+		// TODO check that this still works after the change to the helpers.
+		// maybe we should consider `-1/-1/name` fully valid?
```

Comment:
Yes, `-1/-1` sounds like a good magic number since we already support this under the `sourceIndex` encoding in text.

We could also have a specific constructor for this case.

Reply:
This TODO can be removed, I made a change to `isValid` but then undid it.